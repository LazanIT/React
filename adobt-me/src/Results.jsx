import Pet from "./Pet";

const Results = ({pets}) => {
    return (
        <div className="search">
            {!pets.length ? (
                <h1>No pets found</h1>
            ) : (
                pets.map(pet => {
                    <Pet 
                         // {...pet} // Spread Operator ( Bad idea )
                        animal = {pet.animal}
                        name = {pet.name}
                        breed = {pet.breed}
                        images = {pet.iamges}
                        location = {`${pet.city}, ${pet.state}`}
                        key= {pet.id}
                    />
                })
            )
        }
        </div>
    )

}

export default Results;
