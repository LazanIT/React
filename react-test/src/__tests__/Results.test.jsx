import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Results from "../Results";

test("render correct with no pets", async () => {
  const { asFragment } = render(<Results pets={[]}></Results>);
  expect(asFragment()).toMatchSnapshot();
});
