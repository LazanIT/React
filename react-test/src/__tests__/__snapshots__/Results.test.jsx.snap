// Vitest Snapshot v1

exports[`render correct with no pets 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <h1>
      No Pets Found
    </h1>
  </div>
</DocumentFragment>
`;
